pkg/controller/cluster/cluster_controller.go

```diff
@@ -575,32 +594,52 @@ func (c *Controller) manageMachineSets(machineSets []*clusteroperator.MachineSet
 	return nil
 }
 
-// manageMachineSet determines whether or not a machine set needs to be created because it does not exist, or replaced because it is out of date.
-func (c *Controller) manageMachineSet(cluster *clusteroperator.Cluster, clusterVersion *clusteroperator.ClusterVersion, machineSet *clusteroperator.MachineSet, clusterMachineSetConfig clusteroperator.MachineSetConfig, machineSetNamePrefix string) (*clusteroperator.MachineSet, bool, error) {
+// shouldRecreateMachineSet will return true if the machine set config has changed with
+// the exception of size.
+func shouldRecreateMachineSet(existingConfig, newConfig clusteroperator.MachineSetConfig) bool {
+	existingConfig.Size = 0
+	newConfig.Size = 0
+	return !apiequality.Semantic.DeepEqual(existingConfig, newConfig)
+}
+
+// manageMachineSet determines whether or not a machine set needs to be created because it does not exist,
+// replaced because it is out of date, or updated in case of its size having changed.
+// return values:
+// - machineSet to create (MachineSet)
+// - should update machineset (bool)
+// - should delete machineset (bool)
+// - error
+func (c *Controller) manageMachineSet(cluster *clusteroperator.Cluster, clusterVersion *clusteroperator.ClusterVersion, machineSet *clusteroperator.MachineSet, clusterMachineSetConfig clusteroperator.MachineSetConfig, machineSetNamePrefix string) (*clusteroperator.MachineSet, bool, bool, error) {
 	clusterLog := colog.WithCluster(c.logger, cluster)
 	if machineSet == nil {
 		clusterLog.Debugf("building new machine set")
 		machineSet, err := buildNewMachineSet(cluster, clusterVersion, clusterMachineSetConfig, machineSetNamePrefix)
-		return machineSet, false, err
+		return machineSet, false, false, err
 	}
 
 	msLog := colog.WithMachineSet(clusterLog, machineSet)
 
-	if !apiequality.Semantic.DeepEqual(machineSet.Spec.MachineSetConfig, clusterMachineSetConfig) {
+	if shouldRecreateMachineSet(machineSet.Spec.MachineSetConfig, clusterMachineSetConfig) {
 		msLog.Infof("machine set configuration has changed from %v to %v",
 			machineSet.Spec.MachineSetConfig, clusterMachineSetConfig)
 		machineSet, err := buildNewMachineSet(cluster, clusterVersion, clusterMachineSetConfig, machineSetNamePrefix)
-		return machineSet, true, err
+		return machineSet, false, true, err
 	}
 
 	if machineSet.Spec.ClusterVersionRef.UID != clusterVersion.UID {
 		msLog.Infof("machine set cluster version has changed from %v to %s/%s",
 			machineSet.Spec.ClusterVersionRef, clusterVersion.Namespace, clusterVersion.Name)
 		machineSet, err := buildNewMachineSet(cluster, clusterVersion, clusterMachineSetConfig, machineSetNamePrefix)
-		return machineSet, true, err
+		return machineSet, false, true, err
+	}
+
+	if machineSet.Spec.Size != clusterMachineSetConfig.Size {
+		machineSet.Spec.Size = clusterMachineSetConfig.Size
+		msLog.Infof("machine set size has changed from %d to %d", clusterMachineSetConfig.Size, machineSet.Spec.Size)
+		return nil, true, false, nil
```

Comment:
We need to return the updated machine set so that the controller can use it to send the update machine set to the api server.